src/components/Form/Select.tsx

```diff
@@ -1,143 +1,231 @@
-        input: {
-            borderColor: 'border-gray-500 focus:border-blue-500',
-            color: 'text-gray-900'
-        },
-        text: {
-            color: 'text-gray-900'
-        }
-    }
+const IconStatus = ({ variant }: { variant: SelectVariantType }) => {
+    const { bgIcon, text } = inputVariants[variant]
+    return (
+        <div
+            className={composeClasses('flex justify-center items-center rounded-xl bg-red-50 text-red-500', text.color, bgIcon?.color)}
+            style={{ marginLeft: 11, marginRight: -10, minWidth: 36, minHeight: 36 }}
+            role="defaultIcon"
+        >
+            {variant === 'success' && <CheckCircleIcon aria-label="check" width={24} />}
+            {variant === 'warning' && <InformationCircleIcon aria-label="warning" width={24} />}
+            {variant === 'error' && <XCircleIcon aria-label="error" width={24} />}
+        </div>
+    )
+}
 
-    if (state.success) {
-        classes.input.borderColor = 'border-green-500'
-        classes.text.color = 'text-green-500'
+const getAnimationStyle = (isOpen: boolean): StyleObject => {
+    return {
+        visibility: isOpen ? 'visible' : 'hidden',
+        transition: 'opacity 251ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, transform 167ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+        transform: isOpen ? 'scaleY(1) translateY(0)' : 'scaleY(0.4) translateY(-5px)',
+        opacity: isOpen ? 1 : 0,
+        transformOrigin: '100% 0%'
     }
+}
 
-    if (state.warning) {
-        classes.input.borderColor = 'border-yellow-500'
-        classes.text.color = 'text-yellow-500'
-    }
+function Select({
+    variant = 'default',
+    label,
+    rounded = 'lg',
+    className,
+    classNameAdornment,
+    padding,
+    paddingX = '4',
+    paddingY,
+    startAdornment,
+    endAdornment,
+    message,
+    noBorders,
+    onFocus,
+    onBlur,
+    large,
+    boxShadow = 'lg',
+    style,
+    onChange,
+    optionsList,
+    name,
+    value: selectedValue,
+    ...otherProps
+}: SelectProps) {
+    const getLabel = (selectedValue: any) => {
+        if (!optionsList[`${selectedValue}`]) {
+            return ''
+        }
+
+        if (typeof optionsList[`${selectedValue}`] === 'string') {
+            return optionsList[`${selectedValue}`]
+        }
+
+        const item = optionsList[`${selectedValue}`] as ItemObj
+        if (item.label) return item.label
 
-    if (state.error) {
-        classes.input.borderColor = 'border-red-500'
-        classes.text.color = 'text-red-500'
+        return selectedValue
     }
 
-    if (state.disabled) {
-        classes.input.borderColor = 'border-gray-400 bg-gray-100'
-        classes.input.color = 'text-gray-400'
-        classes.text.color = 'text-gray-900'
+    const [focused, setFocused] = useState(false)
+    const [isOpen, setIsOpen] = useState(false)
+    const [selectedOpt, setSelectedOpt] = useState({
+        value: optionsList[`${selectedValue}`] ? selectedValue : '',
```

Comment:
I defined getLabel outside of the component and pointed directly to the attributes of optionsList using the key, since it's already of type string.
I left this part unchanged because selectedValue may be undefined, and adding an extra validation that I don't see as necessary would be required.

Reply:
Because selectedValue can be undefined and optionsList cannot receive undefined as a key, with a template string, undefined will be converted to a string, and using it as a key will give us 'undefined' and the first condition won't be met. I'm doing it this way to avoid adding an extra validation because I would get the same result.

Reply:
Okey, that's good so merge this pr